server/src/internalClusterTest/java/org/opensearch/action/admin/cluster/stats/ClusterStatsIT.java

```diff
@@ -322,4 +323,81 @@ public void testFieldTypes() {
             }
         }
     }
+
+    public void testNodeCountsWithLegacyMasterSetting() throws ExecutionException, InterruptedException {
```

Comment:
please add test with cluster manager role as well

Reply:
Thanks for checking. Added alongwith refactored some.